backend/worker/kafka_worker.go

```diff
@@ -619,6 +673,7 @@ type KafkaBatchWorker struct {
 	Name                string
 	TracingDisabled     bool
 
-	lastFlush time.Time
-	messages  []kafkaqueue.RetryableMessage
+	lastFlush       time.Time
+	messages        []kafkaqueue.RetryableMessage
+	lastPartitionId *int
```

Comment:
Added `lastPartitionId` to the `KafkaBatchWorker` to use as a filtering key

Reply:
since the flush will has messages across partitions, maybe it makes sense to just flush randomly but sampled? ie. you can have a `flushCounter int64` here and then the `log()` call will log every `N=100?` times?

Reply:
Thats an interesting idea! Correct me if I'm wrong but would that result in random logs being logged - I think one of the benefits of using the `lastPartitionId` is we will get all the logs for a single flush and can see where it may be dropping, vs random logs across multiple flushes